mmvb_backend/benchmarking_sessions/api/views.py

```diff
@@ -0,0 +1,126 @@
+from collections import defaultdict
+
+from django.shortcuts import get_object_or_404
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+
+from benchmarking_sessions.api.serializers import (
+    BenchmarkingSessionResultsSerializer,
+    BenchmarkingSessionSerializer,
+)
+from benchmarking_sessions.models import (
+    BenchmarkingSession,
+    BenchmarkingStepStatus,
+)
+from benchmarking_sessions.tasks import run_benchmark
+from common.utils import CamelCaseAutoSchema
+
+
+# todo: properly document endpoints
+class BenchmarkingSessionViewSet(ModelViewSet):
+    schema = CamelCaseAutoSchema(tags=["BenchmarkingSessions",])
+    serializer_class = BenchmarkingSessionSerializer
+
+    def get_queryset(self):
+        return BenchmarkingSession.objects.all()
+
+    # todo: generate 404
+    @action(methods=["post"], detail=True, url_path="run")
+    def run_benchmark(self, request, *args, **kwargs):
+        benchmarking_session = get_object_or_404(
+            BenchmarkingSession, id=kwargs["pk"]
+        )
+        task = run_benchmark.delay(benchmarking_session.id)
+        benchmarking_session.task_id = task.id
+        benchmarking_session.save()
+        return Response(
+            {
+                "statusUrl": f"/benchmarking-sessions/{benchmarking_session.id}/status"
+            },
+            status=status.HTTP_202_ACCEPTED,
+        )
+
+    # todo: document response structure (OpenAPI)
+    @action(methods=["get"], detail=True, url_path="status")
+    def benchmark_status(self, request, *args, **kwargs):
+        benchmarking_session = get_object_or_404(
+            BenchmarkingSession, id=kwargs["pk"]
+        )
+        if benchmarking_session.status != BenchmarkingSession.Status.RUNNING:
+            return Response(
+                {"status": benchmarking_session.status},
+                status=status.HTTP_200_OK,
+            )
+
+        result = run_benchmark.AsyncResult(benchmarking_session.task_id)
+        if result.status == "PENDING" or result.info is None:
```

Comment:
Maybe it would be a good idea using `BenchmarkingStepStatus` instead of the plain string here.

Reply:
i'll refer to the celery constant instead, which i think is the most correct